### PR TITLE
added files for problems page

### PR DESCRIPTION
**Pull request by fardinzam**
Page for problem description and text editor, still working on it
Screenshot 2023-11-12 at 8 21 09 PM

### CODE REVIEW

**Comment by twang0323** [comment]
image

**Comment by twang0323** [comment]
image

File: src/app/problems/page.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Problems from "@/components/problems/Problems";
+
+const ProblemPg = () => {
```
Comment (shahdivyank) [inline src/app/problems/page.js:3]: call it just Page.js

File: src/components/problems/Description.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { COLORS } from "@/data/Difficulty";
+import { TbBookmark } from "react-icons/tb";
+// import { PROBLEMS} from "@/data/Problem"
```
Comment (shahdivyank) [inline src/components/problems/Description.jsx:6]: remove comments

File: src/components/problems/Description.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        <div className="flex flex-row align-middle">
+          <p>{title}</p>
+          <div className={`${COLORS["Easy"]} text-sm my-auto mx-3`}>
+            {difficulty}
+          </div>
+        </div>
+        <TbBookmark />
+      </div>
+      <div>
+        <div className="rounded-full w-min mx-4 my-2 bg-code-lime">
```
Comment (shahdivyank) [inline src/components/problems/Description.jsx]: this div tag can be integrated into the p tag directly

File: src/components/problems/Description.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      <div className="flex flex-row justify-between mx-4 text-2xl">
+        <div className="flex flex-row align-middle">
+          <p>{title}</p>
+          <div className={`${COLORS["Easy"]} text-sm my-auto mx-3`}>
+            {difficulty}
+          </div>
+        </div>
+        <TbBookmark />
+      </div>
+      <div>
```
Comment (shahdivyank) [inline src/components/problems/Description.jsx:25]: if this div tag needed?

File: src/components/problems/Description.jsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        <div className="rounded-full w-min mx-4 my-2 bg-code-lime">
+          <p className="mx-3 text-code-black text-sm">Operations</p>
+        </div>
+        <div className="mx-4 my-4">
+          {/* {PROBLEMS}.twoSum.objective */}
```
Comment (shahdivyank) [inline src/components/problems/Description.jsx:30]: remove comments

File: src/components/problems/Description.jsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+          Write a program to sum two numbers. Print out the result using the
+          console.log() function. <br />
+          <p className="text-lg text-bold mt-4">Example 1:</p>
```
Comment (shahdivyank) [inline src/components/problems/Description.jsx:33]: use props

File: src/components/problems/Description.jsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+          {/* {PROBLEMS}.twoSum.exampleInput
+              {PROBLEMS}.twoSum.exampleOutput */}
+          <p className="text-lg text-bold mt-4">Constraints:</p>
```
Comment (shahdivyank) [inline src/components/problems/Description.jsx:36]: props

File: src/components/problems/Problems.jsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import TextEditor from "@/components/problems/TextEditor";
+import Description from "@/components/problems/Description";
+// import Submission from "@/components/problems/Submission"
```
Comment (shahdivyank) [inline src/components/problems/Problems.jsx:5]: remove comments

File: src/components/problems/Problems.jsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+      <div className="w-[49%] rounded-md bg-code-darkgray mb-4">
+        <div className="w-1/2 flex flex-row justify-between my-3 ml-5 text-sm">
+          <p className="font-bold">Description</p>
+          <p>Solution</p>
+          <p>Submission</p>
+        </div>
+        <Description title="1. Two Sum" difficulty="Easy" date="Oct 02, 2023" />
+      </div>
+
+      {/* <Submission /> */}
```
Comment (shahdivyank) [inline src/components/problems/Problems.jsx:53]: remove comments

File: src/components/problems/Problems.jsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+import Description from "@/components/problems/Description";
+// import Submission from "@/components/problems/Submission"
+// import { PROBLEMS} from "@/data/Problem"
+
+const Problems = () => {
+  return (
+    <div className="w-11/12 flex flex-row flex-1 justify-between text-white mx-auto">
+      <div className="w-[49%] rounded-md bg-code-darkgray mb-4">
+        <div className="w-1/2 flex flex-row justify-between my-3 ml-5 text-sm">
+          <p className="font-bold">Description</p>
```
Comment (shahdivyank) [inline src/components/problems/Problems.jsx]: props

File: src/components/problems/Problems.jsx [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import TextEditor from "@/components/problems/TextEditor";
+import Description from "@/components/problems/Description";
+// import Submission from "@/components/problems/Submission"
+// import { PROBLEMS} from "@/data/Problem"
+
+const Problems = () => {
+  return (
+    <div className="w-11/12 flex flex-row flex-1 justify-between text-white mx-auto">
+      <div className="w-[49%] rounded-md bg-code-darkgray mb-4">
```
Comment (shahdivyank) [inline src/components/problems/Problems.jsx:17]: make this w-1/2

File: src/components/problems/Problems.jsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+        <div className="w-1/2 flex flex-row justify-between my-3 ml-5 text-sm">
+          <p className="font-bold">Description</p>
+          <p>Solution</p>
+          <p>Submission</p>
+        </div>
+        <Description title="1. Two Sum" difficulty="Easy" date="Oct 02, 2023" />
```
Comment (shahdivyank) [inline src/components/problems/Problems.jsx]: props

File: src/components/problems/Solution.jsx [hunk 13]
```diff
@@ -0,0 +1,34 @@
+// import { COLORS } from "@/data/Difficulty";
+import { FaStar } from "react-icons/fa";
+// import { PROBLEMS} from "@/data/Problem"
```
Comment (shahdivyank) [inline src/components/problems/Solution.jsx:3]: remove comments

File: src/components/problems/Solution.jsx [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+          <p className="align-middle">{title} Solution</p>
+          <div className="flex items-center">
+            <p className="mr-2 text-sm">{date}</p>
+            <FaStar className="text-code-lightorange text-xl" />
+          </div>
+        </div>
+
+        <div>
+          <div className="flex gap-2 mx-4 my-2">
+            <div className="rounded-full bg-code-gray px-3 py-1">
```
Comment (shahdivyank) [inline src/components/problems/Solution.jsx:19]: integrate this div tag directly into p

File: src/components/problems/Solution.jsx [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+              <p className="text-code-white text-sm">Python</p>
+            </div>
+            <div className="rounded-full bg-code-gray px-3 py-1">
```
Comment (shahdivyank) [inline src/components/problems/Solution.jsx:22]: integrate div into p

File: src/components/problems/Solution.jsx [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+              <p className="text-code-white text-sm">C++</p>
+            </div>
+            <div className="rounded-full bg-code-gray px-3 py-1">
```
Comment (shahdivyank) [inline src/components/problems/Solution.jsx:25]: integrate div into p

File: src/components/problems/TextEditor.jsx [hunk 17]
```diff
@@ -0,0 +1,18 @@
+const TextEditor = () => {
+  return (
+    <div className="flex flex-col w-[49%] h-full relative">
+      <div className="w-full h-[85%] border-2 border-code-darkgray rounded-md relative">
```
Comment (shahdivyank) [inline src/components/problems/TextEditor.jsx:4]: integrate div into div

**Comment by shahdivyank** [comment]
> image

Only need support for one language at a time, also make sure it looks like the Figma wireframes

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**